src/xdg.c

```diff
@@ -251,16 +251,36 @@ handle_configure_timeout(void *data)
 	view->pending_configure_serial = 0;
 	view->pending_configure_timeout = NULL;
 
-	/*
-	 * We do not use view_impl_apply_geometry() here since in the timeout
-	 * case we prefer to always put the top-left corner of the view at the
-	 * desired position rather than anchoring some other edge or corner
-	 */
-	bool moved = view->current.x != view->pending.x
-		|| view->current.y != view->pending.y;
-	view->current.x = view->pending.x;
-	view->current.y = view->pending.y;
-	if (moved) {
+	if (view->pending.x != view->current.x
```

Comment:
Hmm. Maybe we should check `wlr_box_empty(&view->pending)` here first. It occurs to me that view->current and view->pending might both be at 0,0 with an initially-fullscreen view.